DevCycle.SDK.Server.Common/API/DVCBaseApiClient.cs

```diff
@@ -22,11 +23,22 @@ public virtual async Task<RestResponse> SendRequestAsync(object json, string url
             request.AddHeader("accept", "application/json");
             request.AddHeader("Authorization", GetServerSDKKey());
 
-            if (queryParams == null) return await restClient.ExecuteAsync(request);
+            if (queryParams == null)
```

Comment:
can this be cleaned up so that rather than early returning and duplicating the `shouldRetry` logic, it just handles the query params if they exist and then does this logic once?